controllers/awsmachine_controller.go

```diff
@@ -428,18 +434,22 @@ func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *
 	switch instance.State {
 	case infrav1.InstanceStatePending, infrav1.InstanceStateStopping, infrav1.InstanceStateStopped:
 		machineScope.SetNotReady()
+		conditions.MarkFalse(machineScope.AWSMachine, infrav1.InstanceReadyCondition, infrav1.InstanceNotReady, clusterv1.ConditionSeverityWarning, "")
 	case infrav1.InstanceStateRunning:
 		machineScope.SetReady()
+		conditions.MarkTrue(machineScope.AWSMachine, infrav1.InstanceReadyCondition)
 	case infrav1.InstanceStateShuttingDown, infrav1.InstanceStateTerminated:
 		machineScope.SetNotReady()
 		machineScope.Info("Unexpected EC2 instance termination", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
 		r.Recorder.Eventf(machineScope.AWSMachine, corev1.EventTypeWarning, "InstanceUnexpectedTermination", "Unexpected EC2 instance termination")
+		conditions.MarkFalse(machineScope.AWSMachine, infrav1.InstanceReadyCondition, infrav1.InstanceTerminated, clusterv1.ConditionSeverityError, "")
 	default:
 		machineScope.SetNotReady()
 		machineScope.Info("EC2 instance state is undefined", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
 		r.Recorder.Eventf(machineScope.AWSMachine, corev1.EventTypeWarning, "InstanceUnhandledState", "EC2 instance state is undefined")
 		machineScope.SetFailureReason(capierrors.UpdateMachineError)
 		machineScope.SetFailureMessage(errors.Errorf("EC2 instance state %q is undefined", instance.State))
+		conditions.MarkFalse(machineScope.AWSMachine, infrav1.InstanceReadyCondition, infrav1.InstanceStateUnknown, clusterv1.ConditionSeverityError, "")
 	}
 
 	// reconcile the deletion of the bootstrap data secret now that we have updated instance state
```

Comment:
I'm wondering if there should be conditions around the creation/deletion of encrypted bootstrap data. Not necessarily something that needs to be addressed in this PR, but this just caught my eye and got me wondering :)

Reply:
Thought a bit about that as well. I think creating encrypted bootstrap data would definitely be useful to track. In case it fails, it can report something more valuable than the current `Info` level `WaitingBootstrapInfo` condition. Deleting the data after consuming it could be valuable but might report `Ready=False` condition on the overall AWSMachine if the delete fails even if the machine is still functioning.